airflow/jobs.py

```diff
@@ -132,6 +132,13 @@ def on_kill(self):
         '''
         pass
 
+    def on_failure(self, e):
```

Comment:
Actually i got this suggestion from this pull request

https://github.com/apache/incubator-airflow/pull/1122#issuecomment-193252883
And @mistercrunch gave me advice to use this call bac